### PR TITLE
Master wowl export state

### PR DESCRIPTION
**Pull request by Polymorphe57**

### CODE REVIEW

File: addons/web/static/src/actions/action_service.js [hunk 1]
```diff
@@ -270,23 +273,36 @@ function makeActionManager(env) {
...
+    }
+
+    if (controller.exportedState) {
+      controller.props.state = controller.exportedState;
+    }
+
     class ControllerComponent extends Component {
       setup() {
         this.Component = controller.Component;
-        this.componentProps = this.props;
```
Comment (Polymorphe57) [inline addons/web/static/src/actions/action_service.js:276]: was not used

File: addons/web/static/src/actions/action_service.js [hunk 2]
```diff
@@ -54,6 +55,11 @@ function makeActionManager(env) {
   // misc
   // ---------------------------------------------------------------------------
 
+  function _getCurrentController() {
```
Comment (aab-odoo) [inline addons/web/static/src/actions/action_service.js:63]: could you add a minimal docstring?

File: addons/web/static/src/actions/action_service.js [hunk 3]
```diff
@@ -858,6 +871,18 @@ function makeActionManager(env) {
       views: controller.views,
       view,
     };
+
+    // LEGACY CODE COMPATIBILITY: remove when controllers will be written in owl
+    if (view.isLegacy) { 
+      if (newController.jsId === controller.jsId) {
+        // case where a legacy view is reloaded via the view switcher
+        const { __legacy_widget__ } = controller.getState();
+        __legacy_widget__.reload();
```
Comment (aab-odoo) [inline addons/web/static/src/actions/action_service.js]: Shouldn't it be:
```suggestion
        return __legacy_widget__.reload();
```

**Review by aab-odoo — APPROVED** [review]
Seems really good :+1:

**Review by aab-odoo — APPROVED** [review]

File: addons/web/static/src/actions/action_service.js [hunk 4]
```diff
@@ -930,6 +959,8 @@ function makeActionManager(env) {
           const viewOptions = {};
           if (state.id) {
             viewOptions.recordId = parseInt(state.id, 10);
+          } else {
```
Comment (Polymorphe57) [inline addons/web/static/src/actions/action_service.js]: Are you ok with that @aab-odoo ?

**Pull request merged**